executor/write.go

```diff
@@ -53,14 +53,15 @@ const (
 //     4. lastInsertID (uint64) : the lastInsertID should be set by the newData.
 //     5. err (error) : error in the update.
 func updateRecord(ctx sessionctx.Context, h int64, oldData, newData []types.Datum, modified []bool, t table.Table,
-	onDup bool) (bool, bool, int64, uint64, error) {
+	onDup, checkReplenish bool) (bool, bool, int64, uint64, error) {
```

Comment:
Can we move the check out of `updateRecord`?
This method is too complex already.

Reply:
Done~